compiler/rustc_typeck/src/check/check.rs

```diff
@@ -705,28 +706,35 @@ fn check_opaque_meets_bounds<'tcx>(
     });
 }
 
-pub fn check_item_type<'tcx>(tcx: TyCtxt<'tcx>, it: &'tcx hir::Item<'tcx>) {
+pub fn check_item_type<'tcx>(tcx: TyCtxt<'tcx>, id: hir::ItemId) {
     debug!(
         "check_item_type(it.def_id={:?}, it.name={})",
-        it.def_id,
-        tcx.def_path_str(it.def_id.to_def_id())
+        id.def_id,
+        tcx.def_path_str(id.def_id.to_def_id())
     );
     let _indenter = indenter();
-    match it.kind {
-        // Consts can play a role in type-checking, so they are included here.
-        hir::ItemKind::Static(..) => {
-            tcx.ensure().typeck(it.def_id);
-            maybe_check_static_with_link_section(tcx, it.def_id, it.span);
-            check_static_inhabited(tcx, it.def_id, it.span);
+    match tcx.hir().def_kind(id.def_id) {
```

Comment:
```suggestion
    match tcx.def_kind(id.def_id) {
```
`tcx.def_kind` is a query, and is cached. `tcx.hir().def_kind` recomputes the value every time, and should be made inaccessible in favor of the query.

Reply:
Sorry, I see there are some I left over. I'll make an update.